extensions/wikia/MultiTasks/MultiDeleteController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare( strict_types=1 );
+
+use Wikia\Logger\Loggable;
+use User;
+
+final class MultiDeleteController extends WikiaController {
+	use Loggable;
+
+	public function allowsExternalRequests() {
+		return false;
+	}
+
+	public function multiDeletePages() {
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
+		if ( !$this->request->wasPosted() ) {
+			$this->response->setCode( 405 );
+			return;
+		}
+		$pagesToDelete = $this->getVal( 'pages' );
+		$userId = $this->getVal( 'user' );
+		$reason = $this->getVal( 'reason' );
+		if ( empty( $pagesToDelete ) || empty( $userId ) ||  empty( $reason ) ) {
+			$this->response->setCode( WikiaResponse::RESPONSE_CODE_BAD_REQUEST );
+			return;
+		}
+		$user = User::newFromId( $userId );
+
+		foreach ( $pagesToDelete as $pageName ) {
+			$page = WikiPage::factory( \Title::newFromText( $pageName ) );
+
+			if ( $page !== null ) {
```

Comment:
Let's make use of the `Loggable` trait and log information about missing pages :)